### PR TITLE
Drop support for CUDA 9.2 & NCCL 2.4

### LINKED ISSUE #5120: [RFC] [v10] Dropping Support for CUDA 9.2 & NCCL 2.4
In CuPy v10, we are thinking of dropping support for CUDA 9.2 and NCCL 2.4. Please leave a comment if you have any concerns.

This does not affect CuPy v9.x releases.

### PR DESCRIPTION
**Pull request by kmaehashi**
Closes #5120.

CUDA 9.2 / NCCL 2.4 specific code removals can be done in separate PRs.

### CODE REVIEW

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
Would you resolve the conflict?

**Comment by kmaehashi** [comment]
Fixed!
pfnCI, test this please.

**Review by takagi — APPROVED** [review]

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docs/source/install.rst [hunk 1]
```diff
@@ -54,11 +54,11 @@ Part of the CUDA features in CuPy will be activated only when the corresponding
 
     * The library to accelerate tensor operations. See :doc:`../reference/environment` for the details.
 
-* `NCCL <https://developer.nvidia.com/nccl>`_: v2.4 (CUDA 9.2) / v2.6 (CUDA 10.0) / v2.8.3 (CUDA 10.1) / v2.8.4 (CUDA 11.1 & 11.2)  / v2.9 (CUDA 10.2 & 11.0)
+* `NCCL <https://developer.nvidia.com/nccl>`_: v2.6 / v2.7 / v2.8 / v2.9
 
     * The library to perform collective multi-GPU / multi-node computations.
 
-* `cuDNN <https://developer.nvidia.com/cudnn>`_: v7.6 (CUDA 9.2 & 10.0) / v8.0 (CUDA 10.1) / v8.1 (CUDA 10.2+) / v8.2 (CUDA 10.2+)
+* `cuDNN <https://developer.nvidia.com/cudnn>`_: v7.6 (CUDA 10.0) / v8.0 (CUDA 10.1) / v8.1 (CUDA 10.2+) / v8.2 (CUDA 10.2+)
```
Comment (kmaehashi) [inline docs/source/install.rst:61]:
```suggestion
* `cuDNN <https://developer.nvidia.com/cudnn>`_: v7.6 / v8.0 / v8.1 / v8.2
```

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Review by takagi — APPROVED** [review]

**Comment by kmaehashi** [comment]
Blocked by https://github.com/chainer/chainer-test/pull/660

**Comment by takagi** [comment]
Would you resolve the conflict?

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**